runtime/src/bank.rs

```diff
@@ -1912,6 +1912,7 @@ impl Bank {
             additional_builtins,
             debug_do_not_add_builtins,
         );
+        bank.fill_missing_sysvar_cache_entries();
```

Comment:
so, this single new line will be most objectionable change in this pr without in-source comments... Argurably, this is touching a important code path.

after some research, in short, i think this is a simple overlook at

Reply:
cc: @Lichtso @jstarry as respective referenced pr authors.

Reply:
> However, I concluded the risk is small overall (use of sysvar cache on snapshot bank is VERY rare as this isn't noticed for months)

I think normal validator operations would never hit this case since we replay and create child banks.

If there were a network outage (plz no) and we needed to restart from a snapshot, would we hit this case and be unable to restart w/o this patch?

Reply:
> If there were a network outage (plz no) and we needed to restart from a snapshot, would we hit this case and be unable to restart w/o this patch?

fortunately, it's not. the pesky hard network outage (hard-fork) case is rigorously tested by local-cluster tests.

as far as i checked, `sysvar_cache` is only used for tx processing. And the snapshot bank itself is frozen/rooted by definition. So, there should be no tx processing. I bet some frozen assert will be triggered, if ever done. 

However, we need populated sysvar_cache for minimized snapshot as a kind of unusual use case here.

So, the other component which is affected would be rpc's tx preflight, with alt tx and with air-gapped node. (this is little known trick, but you can inspect account state at single point of time with rich set of rpc functionality if you make machine stand-alone with empty ledger with single snapshot).

On the other hand, starting to populate sysvar_cache shouldn't cause any bad effects as well. After all, it's rather harmless operation.

Reply:
Got it; it does seem like it was an oversight from the previous PRs on just not including this code on the initialization from fields.

Reply:
Yes you are right. Can't think of any reason why it shouldn't be called here as well. Seems like I simply missed it and it did not break any tests.